source/common/redis/proxy_filter.cc

```diff
@@ -19,9 +19,15 @@ ProxyFilterConfig::ProxyFilterConfig(const Json::Object& config, Upstream::Clust
       cluster_name_(config.getString("cluster_name")),
       stat_prefix_(fmt::format("redis.{}.", config.getString("stat_prefix"))),
       stats_(generateStats(stat_prefix_, scope)) {
-  if (!cm.get(cluster_name_)) {
-    throw EnvoyException(
-        fmt::format("redis filter config: unknown cluster name '{}'", cluster_name_));
+  Upstream::ThreadLocalCluster* cluster = cm.get(cluster_name_);
+  if (cluster == nullptr) {
+    throw EnvoyException(fmt::format("redis filter config: unknown cluster '{}'", cluster_name_));
+  }
+
+  if (cluster->info()->addedViaApi()) {
```

Comment:
OK.. in v2, my thoughts were that clusters that have to live forever get put in https://github.com/lyft/envoy-api/blob/master/api/bootstrap.proto#L25. Everything else is allowed to be dynamically updated, even if its on the filesystem (the file might be replaced, inotify triggers). From that perspective, filters should be capable of fully dynamic behavior in most cases.

How come RDS can't work with a dynamic cluster BTW? Seems there should be no hard dependency there. I can see why CDS needs to be a static cluster :)